internal/rpcserver/rpcwebsocket.go

```diff
@@ -1179,6 +1210,56 @@ func (m *wsNotificationManager) notifyRelevantTxAccepted(tx *dcrutil.Tx,
 	}
 }
 
+// RegisterMixMessages requests notifications to the passed websocket
+// client when mixing messages are accepted to the mixpool.
+func (m *wsNotificationManager) RegisterMixMessages(wsc *wsClient) {
+	select {
+	case m.queueNotification <- (*notificationRegisterMixMessages)(wsc):
+	case <-m.quit:
+	}
+}
+
+// UnregisterMixMessages stops notifications to the websocket client of any
+// newly-accepted mixing messages.
+func (m *wsNotificationManager) UnregisterMixMessages(wsc *wsClient) {
+	select {
+	case m.queueNotification <- (*notificationUnregisterMixMessages)(wsc):
+	case <-m.quit:
+	}
+}
+
+// notifyMixMessage notifies all clients subscribed to mixing messages with
+// the accepted mixing message.
+func (m *wsNotificationManager) notifyMixMessage(clients map[chan struct{}]*wsClient,
+	msg mixing.Message) {
+
+	// Skip notification creation if no clients have requested mix
+	// notifications.
+	if len(clients) == 0 {
+		return
+	}
+
+	// Write write message payload in hex encoding.
+	buf := new(bytes.Buffer)
+	err := msg.BtcEncode(hex.NewEncoder(buf), wire.MixVersion)
+	if err != nil {
+		// Should never error; the message has already been processed
+		// and accepted.
+		panic(err)
```

Comment:
I know this is true, but let's not take down the whole process if it somehow manages to happen.